ykrt/src/deopt.rs

```diff
@@ -143,37 +148,35 @@ extern "C" fn __llvm_deoptimize(
     }
 }
 
-/// Called when a guard failure occurs. Handles the reading of stackmaps, matching of JIT to AOT
-/// variables, etc., in order to reconstruct the stack.
-#[cfg(target_arch = "x86_64")]
-#[no_mangle]
-unsafe extern "C" fn __ykrt_deopt(
-    ctr: *const CompiledTrace,
-    // Address of the control point's frame.
+// Struct storing information we need to pass via the `LLVMOrcThreadSafeModuleWithModuleDo`
+// function.
+struct ReconstructInfo<'a> {
+    ctr: Arc<CompiledTrace>,
     frameaddr: *mut c_void,
-    // Struct describing the location of the AOT live variables.
-    aotvals: &LiveAOTVals,
-    // Address and size of vector holding active AOT frame information needed to recreate them.
-    actframes: &CVec,
-    // ID of the failing guard.
-    guardid: usize,
-    // Return address of deoptimize call. Used to find the correct stackmap record.
+    aotvals: &'a LiveAOTVals,
+    actframes: &'a CVec,
     retaddr: usize,
-    // Current stack pointer. Needed to read spilled register values from the stack.
     rsp: *const c_void,
-) -> NewFramesInfo {
-    #[cfg(feature = "yk_jitstate_debug")]
-    print_jit_state("deoptimise");
-
-    // Put the CompiledTrace back into an Arc, so it is dropped properly.
-    let ctr = Arc::from_raw(ctr);
+    nfi: Option<NewFramesInfo>,
+}
 
-    // FIXME: Check here if we have a side trace and execute it. Otherwise just increment the guard
-    // failure counter.
+// Function containing the actual code setting up the frame reconstructor. This function will be
```

Comment:
I think we can assume this is a function :) I guess "Reconstruct the stack frames for deopt" or similar is a clearer first sentence?

Reply:
Fixed in [cc9ce02](https://github.com/ykjit/yk/pull/780/commits/cc9ce02bb6df86ce89ffa838fd2996aef13b5029)